src/class/api.php

```diff
@@ -0,0 +1,356 @@
+            //  and client didn't add */* as an option. So, we complain.
+            $this->sFormatOutput = $this->aFormatsAccepted[0]; // Pick our default output.
+            $this->aResponse['errors'][] = 'The format you requested is not available. Pick from ' . implode(', ', $this->aFormatsAccepted);
+            $this->sendHeader(406, true); // Send 406 Not Acceptable, print response, and quit.
+
+        } elseif ($aAccepts) {
+            // Client provided request, and we can match.
+            // We'll loop through our preferred formats, and pick what works best with the request.
+            if (in_array('*/*', $aAccepts)) {
+                // Client is OK with everything.
+                $this->sFormatOutput = $this->aFormatsAccepted[0]; // Pick our default output.
+            } else {
+                foreach ($this->aFormatsAccepted as $sFormat) {
+                    // Take the prefix, as we'll accept 'type/*' as well.
+                    $sType = substr($sFormat, 0, strpos($sFormat . '/', '/'));
+                    if (in_array($sFormat, $aAccepts) || in_array($sType . '/*', $aAccepts)) {
+                        // Direct match or match on type.
+                        $this->sFormatOutput = $sFormat;
+                        break;
+                    }
+                }
+            }
+
+        } else {
+            // Client didn't bother to request anything.
+            $this->sFormatOutput = $this->aFormatsAccepted[0]; // Pick our default output.
+        }
+
+
+
+
+
+        // Parse the URL, to see what we're requested to do.
+        // To prevent notices.
+        $_PE = array_pad($_PE, 5, '');
+
+        // $_PE[0] should always be API.
+        if (strtolower($_PE[0]) != 'api') {
+            $this->aResponse['errors'][] = 'Could not parse requested URL.';
+            $this->sendHeader(400, true); // Send 400 Bad Request, print response, and quit.
+        }
+
+        // Check if we're using the old style LOVD2-style API or not.
+        if (in_array($_PE[1], array('rest', 'rest.php'))) {
+            // Yes, we are...
+            $this->aResponse['version'] = 0;
+            // This API also ignores the Accept header.
+            $this->sFormatOutput = 'text/plain';
+            // And, we only allow GET.
+            if (!GET) {
+                // Will only allow GET.
+                // $this->aResponse['errors'][] = 'Method not allowed here.';
+                // $this->sendHeader(405, true); // Send 405 Method Not Allowed, print response, and quit.
+                // This API is LOVD2-style and shouldn't have their output changed now that we're more advanced.
+                header('HTTP/1.0 501 Not Implemented');
+                exit;
+            }
+
+            // Parse URL to see what we need to do.
+            list(,,
+                $this->sResource, // 2
+                $this->sGene,     // 3
+                $this->nID) = $_PE;
+
```

Comment:
It might be an idea to move the unpacking of $_PE to line 141, so indexing $_PE is not necessary anymore in the code above.